libraries/node-core-library/src/test/Executable.test.ts

```diff
@@ -298,17 +284,44 @@ describe('Executable process tests', () => {
       currentWorkingDirectory: executableFolder
     });
     const result: IWaitForExitResult<string> = await Executable.waitForExitAsync(childProcess, {
-      encoding: 'utf8',
-      throwOnNonZeroExitCode: false
+      encoding: 'utf8'
     });
     expect(result.exitCode).toEqual(1);
+    expect(result.signal).toBeNull();
     expect(typeof result.stdout).toEqual('string');
     expect(typeof result.stderr).toEqual('string');
-    expect(result.stdout.indexOf('Executing javascript-file.js with args:')).toEqual(-1);
+    expect(result.stdout).toMatch(/^Executing npm-binary-wrapper(\.cmd)? with args:/);
     expect(result.stderr.endsWith('This is a failure'));
   });
 
-  test('Executable.runToCompletion(Executable.spawn("npm-binary-wrapper")) failure with throw', async () => {
+  test('Executable.runToCompletion(Executable.spawn("no-terminate")) killed', async () => {
+    const executablePath: string = path.join(executableFolder, 'no-terminate', 'javascript-file.js');
+    const childProcess: child_process.ChildProcess = Executable.spawn(
+      process.argv0,
+      [executablePath, '1', '2', '3'],
+      {
+        environment,
+        currentWorkingDirectory: executableFolder
+      }
+    );
+    const startTime: number = Date.now();
+    await Async.sleepAsync(50); // Wait for the process to boot and print
```

Comment:
Should we wait for the line to be printed by listening to the stdout?

Reply:
If it becomes a problem, we'll change this.